EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInput.java

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ *
+ * Copyright (c) 2012-2017, Business Process Technology (BPT),
+ * http://bpt.hpi.uni-potsdam.de.
+ * All Rights Reserved.
+ *
+ *******************************************************************************/
+package de.hpi.unicorn.application.pages.input.generator.attributeInput;
+
+import de.hpi.unicorn.application.pages.input.generator.validation.AttributeValidator;
+import de.hpi.unicorn.event.attribute.AttributeTypeEnum;
+import de.hpi.unicorn.event.attribute.TypeTreeNode;
+import org.apache.log4j.Logger;
+import org.apache.wicket.validation.IValidator;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public abstract class AttributeInput implements Serializable {
+	private final TypeTreeNode attribute;
+	private String input = "";
+	static Random random = new Random();
+
+	final List<ProbabilityDistributionEnum> availableMethods = new ArrayList<>();
+
+	private ProbabilityDistributionEnum selectedMethod;
+
+	static final Logger logger = Logger.getLogger(AttributeInput.class);
+
+	AttributeInput(TypeTreeNode attribute) {
+		this.attribute = attribute;
+	}
+
+	public static AttributeInput attributeInputFactory(TypeTreeNode attribute) {
+		switch (attribute.getType()) {
+			case STRING:
+				return new StringAttributeInput(attribute);
+			case INTEGER:
+				return new IntegerAttributeInput(attribute);
+			case FLOAT:
+				return new FloatAttributeInput(attribute);
+			case DATE:
+				return new DateAttributeInput(attribute);
+			default:
+				attribute.setType(AttributeTypeEnum.STRING);
+				return new StringAttributeInput(attribute);
+		}
+	}
+
+	public TypeTreeNode getAttribute() {
+		return attribute;
+	}
+
+	public AttributeTypeEnum getAttributeType() {
+		return this.getAttribute().getType();
+	}
+
+	public String getAttributeName() {
+		return this.getAttribute().getName();
+	}
+
+	public List<ProbabilityDistributionEnum> getAvailableMethods() {
+		return availableMethods;
+	}
+
+	public ProbabilityDistributionEnum getSelectedMethod() {
+		return selectedMethod;
+	}
+
+	public void setSelectedMethod(ProbabilityDistributionEnum selectedMethod) {
+		this.selectedMethod = selectedMethod;
+	}
+
+	public String getInput() {
+		return input;
+	}
+
+	String getInputOrDefault() {
+		if(input != null && !input.isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') 'if' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)